setuptools_scm/__init__.py

```diff
@@ -37,6 +37,7 @@ def _version_from_entrypoint(root, entrypoint):
 
 
 def dump_version(root, version, write_to, template=None):
+    assert isinstance(version, string_types)
```

Comment:
the code in dump-version only works with strings, perhaps even this check is not trough enough